middleware/wrap_writer.go

```diff
@@ -81,6 +81,10 @@ type basicWriter struct {
 }
 
 func (b *basicWriter) WriteHeader(code int) {
+	if code >= 100 && code <= 199 && code != http.StatusSwitchingProtocols {
+		b.ResponseWriter.WriteHeader(code)
```

Comment:
I think we're missing `if !b.discard {`, same as below

Reply:
ah, you're right! Fixed. PTAL